library/core/qsl_base/src/main/java/org/quiltmc/qsl/base/mixin/SharedConstantsMixin.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.base.mixin;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+import net.minecraft.SharedConstants;
+
+import net.fabricmc.loader.api.FabricLoader;
+
+@Mixin(SharedConstants.class)
+public class SharedConstantsMixin {
+	@Shadow
+	public static boolean isDevelopment = FabricLoader.getInstance().isDevelopmentEnvironment();
```

Comment:
My one cent on this:
1) This should use Quilt's method for that
2) oh god no oh god no oh god no oh god no oh god no oh god no
While it would be great to toggle few of the goodies on the dev mode, there are few bits that are only useful for Mojang and that would actually be harmful to us; So, yeah, we actually need a proper PR for this, not a sneaked-in feature from an unrelated PR